Makefile

```diff
@@ -100,19 +100,21 @@ download-sidecar-executable:
 ifeq ($(SKIP_DOWNLOAD_EXECUTABLE),true)
 	@echo "Skipping download of sidecar executable since it already exists at $(EXECUTABLE_DOWNLOAD_PATH)"
 else
-ifeq ($(CI),true)
-	mkdir -p bin && \
-	export EXECUTABLE_PATH=ide-sidecar-$(IDE_SIDECAR_VERSION_NO_V)-runner-$(SIDECAR_OS_ARCH) && \
-		gh release download $(IDE_SIDECAR_VERSION) --repo $(IDE_SIDECAR_REPO) --pattern=$${EXECUTABLE_PATH} --output $(EXECUTABLE_DOWNLOAD_PATH) --clobber && \
-		chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
-		echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"
-else
-	mkdir -p bin && \
-	export EXECUTABLE_PATH=ide-sidecar-$(IDE_SIDECAR_VERSION_NO_V)-runner-$(SIDECAR_OS_ARCH) && \
-		curl -L -o $(EXECUTABLE_DOWNLOAD_PATH) "https://github.com/$(IDE_SIDECAR_REPO)/releases/download/$(IDE_SIDECAR_VERSION)/$${EXECUTABLE_PATH}" && \
-		chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
-		echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"
-endif
+	@if ! [ "$$CI" = "true" ] || [[ "$$SEMAPHORE_ORGANIZATION_URL" == *".semaphoreci.com" ]]; then \
+		mkdir -p bin && \
+		echo "Using curl to download sidecar executable from GitHub release $(IDE_SIDECAR_VERSION)"; \
+		export EXECUTABLE_PATH=ide-sidecar-$(IDE_SIDECAR_VERSION_NO_V)-runner-$(SIDECAR_OS_ARCH) && \
+			curl -L -o $(EXECUTABLE_DOWNLOAD_PATH) "https://github.com/$(IDE_SIDECAR_REPO)/releases/download/$(IDE_SIDECAR_VERSION)/$${EXECUTABLE_PATH}" && \
+			chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
+			echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"; \
+	else \
+		mkdir -p bin && \
+		echo "Using gh to download sidecar executable from GitHub release $(IDE_SIDECAR_VERSION)"; \
+		export EXECUTABLE_PATH=ide-sidecar-$(IDE_SIDECAR_VERSION_NO_V)-runner-$(SIDECAR_OS_ARCH) && \
+			gh release download $(IDE_SIDECAR_VERSION) --repo $(IDE_SIDECAR_REPO) --pattern=$${EXECUTABLE_PATH} --output $(EXECUTABLE_DOWNLOAD_PATH) --clobber && \
+			chmod +x $(EXECUTABLE_DOWNLOAD_PATH) && \
+			echo "Downloaded sidecar executable to $(EXECUTABLE_DOWNLOAD_PATH)"; \
+	fi
```

Comment:
I get that you just changed the indentation of these lines and modified the condition on line 103, but why do we need to use two different techniques to download from the ide-sidecar releases?

Can we just use one of these approaches?

Reply:
My rationale was that if we could use the authenticated `gh` CLI to issue the download request, then we should do so, in order to leverage a higher tier of GitHub API performance and rate limiting. (I'm not sure if this is actually the case).

Having said that, I would be fine with simplifying down to just the `curl` approach. The change can be made in a follow-up PR, like you suggested. Thanks!

Reply:
Yeah, might be worth it to simplify the logic in a future PR. Thanks!